launch/launch/actions/execute_process.py

```diff
@@ -257,14 +257,30 @@ def _parse_cmdline(
         :returns: a list of command line arguments.
         """
         args = []
-        for arg in parser.parse_substitution(cmd):
-            if isinstance(arg, TextSubstitution):
-                text = arg.text
-                text = shlex.split(text)
-                text = [TextSubstitution(text=item) for item in text]
-                args.extend(text)
+        arg = []
+        for sub in parser.parse_substitution(cmd):
+            if isinstance(sub, TextSubstitution):
+                some_args = shlex.split(sub.text)
+                if not some_args:
+                    continue
+                if sub.text[0].isspace():
```

Comment:
I think a comment describing this if-block would be good. Same for the other blocks below.

Reply:
I added comments in all the conditions. I'm not sure if they clarify something, or if they make the code more obscure :rofl:.